File: src/main/frontend/src/service/CargoService.ts
import {Page} from "../model/Page";
import {Cargo} from "../model/Cargo";
import {PageParams} from "../model/PageParams";
import {CargoDetailsDto} from "../model/CargoDetailsDto";

export interface CargoService {
    getCargos(page: PageParams): Promise<Page<Cargo>>;
    getCargo(id?: number): Promise<Cargo>;
    getCargoDetails(id?: number): Promise<CargoDetailsDto>;
    saveCargo(cargo: Cargo): Promise<Cargo>;
}

File: src/main/java/com/raihanorium/springreact/service/CargoService.java
package com.raihanorium.springreact.service;

import com.raihanorium.springreact.dto.CargoDetailsDto;
import com.raihanorium.springreact.model.Cargo;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface CargoService {
    Optional<Cargo> findById(Long id);

    Optional<Cargo> findByName(String cargoName);

    Page<Cargo> findAll(String name, Pageable pageable);

    Cargo save(Cargo cargo);

    void deleteAll();

    CargoDetailsDto getCargoDetails(Long cargoId);
}
